src/app/child-dev-project/children/child-select/child-select.component.ts

```diff
@@ -21,56 +22,41 @@ import { MatAutocompleteTrigger } from "@angular/material/autocomplete";
 export class ChildSelectComponent implements OnChanges {
   searchText = "";
   showOnlyActiveChildren: boolean = true;
-  suggestions = new Array<Child>();
-  allChildren = new Array<Child>();
-  selectedChildren = new Array<Child>();
+  suggestions: Child[] = [];
+  allChildren: Child[] = [];
+  selectedChildren: Child[] = [];
 
-  @Input() valueAsIds: string[];
+  @Input() selectedChildrenIds: string[] = [];
+  @Output() selectedChildrenIdsChange = new EventEmitter<string[]>();
   @Input() disabled: boolean;
-  @Output() valueAsIdsChange = new EventEmitter();
-  @Output() newIdAdded = new EventEmitter();
-  @Output() idRemoved = new EventEmitter();
+  @Output() newIdAdded = new EventEmitter<string>();
+  @Output() idRemoved = new EventEmitter<string>();
 
-  @ViewChild("inputField", { static: true }) inputField;
+  @ViewChild("inputField", { static: true })
+  inputField: ElementRef<HTMLInputElement>;
   @ViewChild(MatAutocompleteTrigger) autocomplete: MatAutocompleteTrigger;
 
   constructor(private childrenService: ChildrenService) {}
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes.hasOwnProperty("valueAsIds")) {
-      this.loadInitial();
+    if (changes.hasOwnProperty("selectedChildrenIds")) {
+      this.childrenService
```

Comment:
any particular reason to inline this rather than keeping it in explicitly name sub-methods?

Reply:
I wanted to keep it similar to the similar to the new SchoolSelectComponent so it is easier to combine the two components